src/sentry/middleware/user.py

```diff
@@ -1,17 +1,19 @@
 from __future__ import absolute_import
 
-from django.utils import timezone
 from datetime import timedelta
-import inspect
 
+from django.utils import timezone
+from sentry.api.authentication import TokenAuthentication
 
-class UserActiveMiddleware(object):
-    allowed_paths = (
-        'sentry.api.endpoints',
-        'sentry.web.frontend',
-    )
 
+class UserActiveMiddleware(object):
     def process_view(self, request, view_func, view_args, view_kwargs):
+        try:
+            auth = TokenAuthentication()
+            request.user = auth.authenticate(request)[0]
```

Comment:
I don't understand the need for case 2—the auth.py middleware fires before the SessionAuthentication and if we're not looking to handle TokenAuth, I don't see what requests would not have a user in case 1, but have a user for case 2.